static/js/game.js

```diff
@@ -223,33 +248,57 @@ function GameController(canvas) {
         }
     };
 
-    this.updateUnit = function (unitData) {
-        var id = unitData['id'];
-
+    this.updateProperties = function(unit, unitData) {
         for (var key in unitData) {
-            if (this.units[id].hasOwnProperty(key)) {
-                this.units[id][key] = unitData[key]
+            if (unit.hasOwnProperty(key)) {
+                unit[key] = unitData[key]
 
             }
             else if (key = 'angle') {
-                this.units[id]['rotation'] = unitData[key]
+                unit['rotation'] = unitData[key]
             }
         }
-        this.units[id].regX = this.units[id].width / 2;
-        this.units[id].regY = this.units[id].height / 2;
-        this.units[id].speedTick = this.units[id].speed / this.frequency / FPS;
-        if (this.hero && id == this.hero['id']){
-            this.updateTableScorecards()
-        };
-        this.updateScorecardHeroes(unitData);
+        unit.regX = unitData.width / 4;
+        unit.regY = unitData.height / 4;
+        return unit
+    };
+
+    this.updateUnit = function (unitData) {
+        var id = unitData['id'];
+
+        if (this.units[id]) {
+            this.units[id] = this.updateProperties(this.units[id], unitData);
+            this.units[id].speedTick = this.units[id].speed / this.frequency / FPS;
+            if (id == this.hero['id']){
+                this.updateTableScorecards()
+            };
+            this.updateScorecardHeroes(unitData);
+
+        }
     };
 
     this.killUnit = function (unitData) {
         var id = unitData['id'];
-        this.stage.removeChild(this.units[id]);
-        this.stage.update();
-        this.updateScorecardHeroes(unitData);
-        delete this.units[id];
+        if (this.units[id]) {
+            var animation = this.setAnimation(this.units[id]);
+            var unit = new createjs.Sprite(animation, "died");
+            unit = this.updateProperties(unit, unitData);
+            this.stage.removeChild(this.units[id]);
```

Comment:
I think you can use your own method called `deleteUnit` here

e.g.:
deleteUnit(this.units[id])